src/plugins/workspace/server/saved_objects/workspace_ui_settings_client_wrapper.ts

```diff
@@ -0,0 +1,122 @@
+   * WORKSPACE_TYPE needs to be excluded.
+   */
+  private getWorkspaceTypeEnabledClient(request: OpenSearchDashboardsRequest) {
+    return this.getScopedClient?.(request, {
+      includedHiddenTypes: [WORKSPACE_TYPE],
+      excludedWrappers: [WORKSPACE_UI_SETTINGS_CLIENT_WRAPPER_ID],
+    }) as SavedObjectsClientContract;
+  }
+
+  public setScopedClient(getScopedClient: SavedObjectsServiceStart['getScopedClient']) {
+    this.getScopedClient = getScopedClient;
+  }
+
+  public wrapperFactory: SavedObjectsClientWrapperFactory = (wrapperOptions) => {
+    const getUiSettingsWithWorkspace = async <T = unknown>(
+      type: string,
+      id: string,
+      options: SavedObjectsBaseOptions = {}
+    ): Promise<SavedObject<T>> => {
+      const { requestWorkspaceId } = getWorkspaceState(wrapperOptions.request);
+
+      /**
+       * When getting ui settings within a workspace, it will combine the workspace ui settings with
+       * the global ui settings and workspace ui settings have higher priority if the same setting
+       * was defined in both places
+       */
+      if (type === 'config' && requestWorkspaceId) {
+        const configObject = await wrapperOptions.client.get<Record<string, any>>(
+          'config',
+          id,
+          options
+        );
+
+        const workspaceObject = await this.getWorkspaceTypeEnabledClient(
+          wrapperOptions.request
+        ).get<WorkspaceAttribute>(WORKSPACE_TYPE, requestWorkspaceId);
+
+        configObject.attributes = {
+          ...configObject.attributes,
+          ...workspaceObject.attributes.uiSettings,
+        };
+
+        return configObject as SavedObject<T>;
+      }
+
+      return wrapperOptions.client.get(type, id, options);
+    };
+
+    const updateUiSettingsWithWorkspace = async <T = unknown>(
+      type: string,
+      id: string,
+      attributes: Partial<T>,
+      options: SavedObjectsUpdateOptions = {}
+    ): Promise<SavedObjectsUpdateResponse<T>> => {
+      const { requestWorkspaceId } = getWorkspaceState(wrapperOptions.request);
+
+      /**
+       * When updating ui settings within a workspace, it will update the workspace ui settings,
+       * the global ui settings will remain unchanged.
+       */
+      if (type === 'config' && requestWorkspaceId) {
+        const configObject = await wrapperOptions.client.get<Record<string, any>>(
+          'config',
+          id,
+          options
+        );
+        const savedObjectsClient = this.getWorkspaceTypeEnabledClient(wrapperOptions.request);
+
+        const workspaceObject = await savedObjectsClient.get<WorkspaceAttribute>(
+          WORKSPACE_TYPE,
+          requestWorkspaceId
+        );
+
+        const workspaceUpdateResult = await savedObjectsClient.update<WorkspaceAttribute>(
+          WORKSPACE_TYPE,
+          requestWorkspaceId,
+          {
+            ...workspaceObject.attributes,
+            uiSettings: { ...workspaceObject.attributes.uiSettings, ...attributes },
```

Comment:
Since uiSettings may include some not exists properties in passed attributes. Shall we need to overwrite workspace uiSettings with passed attributes directly? Or it will be hard to delete properties.

Reply:
You mean we only accept what's needed by workspace? like `defaultIndex` for now?

Reply:
I mean modify code like this:
```ts

        const workspaceUpdateResult = await savedObjectsClient.update<WorkspaceAttribute>(
          WORKSPACE_TYPE,
          requestWorkspaceId,
          {
            ...workspaceObject.attributes,
            uiSettings: attributes,
          }
```
Is there any remove config property cases? It seems we can only set property to `undefined` in current implementation. We can't delete specific properties in global config or `workspace.attributes.uiSettings`.

Reply:
I see, but ui settings is normally used like this: `uiSettings.set('defaultIndex', 'value')`, the `attributes` here is always partial object.

Reply:
For most cases, the full attributes was passed to savedObjectClient.updated. Do you mean we will call `savedObjects.client.update('config', id, {'defaultIndex', 'value'})` directly? It seems will remove all other attributes when workspace disabled.